src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -1210,11 +1210,11 @@ internal void ThrowInvalidDataContractException(string message)
         [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
         internal static bool IsTypeSerializable(Type type)
         {
-            return IsTypeSerializable(type, new HashSet<Type>());
+            return IsTypeSerializable(type, null);
         }
 
         [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
-        private static bool IsTypeSerializable(Type type, HashSet<Type> previousCollectionTypes)
+        private static bool IsTypeSerializable(Type type, HashSet<Type>? previousCollectionTypes)
```

Comment:
If you make this internal and add ` = null` to the `previousCollectionTypes` parameter, then you can delete the above overload.